lib/IRGen/GenCall.cpp

```diff
@@ -2605,9 +2606,16 @@ class SyncCallEmission final : public CallEmission {
     if (fnConv.getNumDirectSILResults() == 1
         && (fnConv.getDirectSILResults().begin()->getConvention()
             == ResultConvention::Autoreleased)) {
-      if (IGF.IGM.Context.LangOpts.EnableObjCInterop)
-        result = emitObjCRetainAutoreleasedReturnValue(IGF, result);
-      else
+      if (IGF.IGM.Context.LangOpts.EnableObjCInterop) {
+        auto ty = fnConv.getSILResultType(IGF.IGM.getMaximalTypeExpansionContext());
+        auto *classTypeInfo = dyn_cast<ClassTypeInfo>(&IGF.IGM.getTypeInfo(ty));
```

Comment:
FYI this will succeed for any `ReferenceTypeInfo`, I have a patch to fix crashes here: https://github.com/apple/swift/pull/73915

Reply:
Thank you for the fix @hamishknight.